src/readme.md

```diff
@@ -0,0 +1,140 @@
+# Developer notes
```

Comment:
I think this file is generally named CONTRIBUTING.md on github shall we just use that?

Reply:
Done, thanks.